### ISSUE TITLE
pin_memory when learning

### ISSUE THREAD

**Issue by nickluijtgaarden** [ChenRocks/fast_abs_rl#52]
Hey Chen,

I have a question about the DataLoader in the files where the extractor, abstractor and RL are trained, more specifically, about this line of code:

  val_loader = DataLoader(
        MatchDataset('val'), batch_size=BUCKET_SIZE,
        shuffle=False, num_workers=4 if cuda and not debug else 0,
        collate_fn=coll_fn
    )

I was wondering why pin_memory is not set to TRUE? I have read that this can improve performance when training on Nvidia GPU's. Also, would you advise to set num_workers higher when using a high-end GPU (Tesla V100 16GB) and 8 cores CPU?

Thanks!

- Nick

**Comment by ChenRocks** [comment]
I'm not familiar with the pin_memory option. Is there any benchmark on performance? I remember I get some error late in training using the option but that was long time ago...

During the initial development of this project I only have access to K-80 and I think the current configuration is at reasonable performance. I don't have a benchmark though but I don't think the bottleneck is the number of CPU cores.

Let me know the results if you tried any of these :)
